flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/functions/hive/conversion/HiveInspectors.java

```diff
@@ -162,34 +220,64 @@ public static Object toFlinkObject(ObjectInspector inspector, Object data) {
 
 				return oi.preferWritable() ?
 					oi.get(data) :
-					oi.getPrimitiveWritableObject(data);
+					oi.getPrimitiveJavaObject(data);
```

Comment:
I'm not sure if we need to call oi.preferWritable() because we are converting Hive data to Flink data. Flink is always expecting java object. Thus, oi.getPrimitiveJavaObject() might be enough. I'm not 100% sure, though.

Reply:
that's a good point. Let me read Hive source code one more time for this

Reply:
I read the source code and did some more testing. HiveSimpleUDF and HiveGenericUDF don't need writables as we explicitly requires primitive return type in both of them. I'm not fully sure about UDTF and UDAF as we don't have them yet, nor does tests for them. 

I recommend keep them for now as they are not doing any harms. I've created "[FLINK-12891] evaluate converting hadoop/hive writable between Hive functions and Flink" to re-evaluate this issue once we have support for all hive functions in place in 1.9.

Reply:
Okay. I took a little deeper look and believe we only need to call getPrimitiveJavaObject() to get the java object, which can be further converted to the object that Flink expects. Even if we call getPrimitiveWritableObject() to get a writable object, we still need to convert to its java equivalent and then further convert it to the java object that Flink expects.

However, this can be taken care of in a separate PR.